File: 0069-sqrtx/0069-sqrtx.c
int mySqrt(int x) {
    int low = 1, high = x, ans = 0;
    while(low <= high){
        int mid = low + (high - low) / 2;
        if(mid == x/mid){
            return mid;
        } else if (mid > x / mid){
            high = mid - 1;
        } else {
            low = mid + 1;
            ans = mid;
        }
    }
    return ans;
}

File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        int left = 1, right = x, sqrt = 0;
        while(left <= right){
            int mid = left + (right - left) / 2;
            if(mid == x/mid){
                return mid;
            } else if (mid > x / mid){
                right = mid - 1;
            } else {
                left = mid + 1;
                sqrt = mid;
            }
        }
        return sqrt;
    }
}